### PR TITLE
Fix LocalRunner hanging for BatchSamplers

### PR DESCRIPTION
**Pull request by gautams3**
Set my tf session to use a config
https://github.com/tensorflow/tensorflow/issues/14442#issuecomment-343501993

### CODE REVIEW

**Comment by naeioi** [comment]
Nice catch. Could you provide a case to show this is useful? 

Also, this PR should only be able to fix garage for multithreading, not multiprocessing, as pointed out in https://github.com/tensorflow/tensorflow/issues/14442#issuecomment-343501993.

File: garage/experiment/local_tf_runner.py [hunk 1]
```diff
@@ -66,7 +66,7 @@ def __init__(self, sess=None, max_cpus=1):
         if max_cpus > 1:
             from garage.sampler import singleton_pool
             singleton_pool.initialize(max_cpus)
-        self.sess = sess or tf.Session()
+        self.sess = sess or tf.Session(config=tf.ConfigProto(use_per_session_threads=True))
```
Comment (ryanjulian) [inline garage/experiment/local_tf_runner.py]: @naeioi  I think this is testable.

@gautams3 can you provide a minimal launcher which reproduces the problem?

Comment (gautams3) [reply garage/experiment/local_tf_runner.py]: This needs a sampler and a LocalRunner for the sampler to call `start_worker`. One option is the [point_reach_embed launcher in embed2learn](https://github.com/ryanjulian/embed2learn-private/blob/388934023af752f5137791319f26164bebdae256/launchers/point_reach_embed.py)

Comment (naeioi) [reply garage/experiment/local_tf_runner.py]: Could you grant me access to embed2learn? I cannot open the link above.

Comment (ryanjulian) [reply garage/experiment/local_tf_runner.py]: done

**Comment by ryanjulian** [comment]
@gautams3 can you rebase this on master and make it pass the CI?

**Comment by ryanjulian** [comment]
@naeioi can you get this submitted?

**Comment by naeioi** [comment]
Closing this because the issue of freezing of batch sampler is already handled in the local runner. See https://github.com/rlworkgroup/garage/blob/master/garage/experiment/local_tf_runner.py#L54-L63.
I tested embed2learn and didn't see it stuck. Gautam and I discussed on this PR and we concluded that it won't fix anything.

**Pull request closed without merging**